educator/src/Dscp/Educator/Web/Server.hs

```diff
@@ -66,16 +68,17 @@ createGetStudentsAction = do
     traverse_ addKeyWithSeed [1000..1100]
     return . GetStudentsAction $ atomically . readTVar $ tvr
 
-serveStudentAPIReal :: HasEducatorConfig => EducatorWebParams -> EducatorRealMode ()
-serveStudentAPIReal EducatorWebParams{..} = do
+serveEducatorAPIsReal :: CombinedWorkMode ctx m => EducatorWebParams -> m ()
+serveEducatorAPIsReal EducatorWebParams{..} = do
     let ServerParams{..} = ewpServerParams
     getStudents <- liftIO $ createGetStudentsAction
     let srvCtx = getStudents :. EmptyContext
 
     logInfo $ "Serving Student API on "+|spAddr|+""
-    eCtx <- ask
-    let educatorApiServer = mkEducatorApiServer (convertEducatorApiHandler eCtx)
-    studentApiServer <- mkStudentApiServer (convertStudentApiHandler eCtx) ewpBotParams
+    unliftIO <- askUnliftIO
+    let educatorApiServer = mkEducatorApiServer (convertEducatorApiHandler unliftIO)
+    studentApiServer <- mkStudentApiServer (convertStudentApiHandler unliftIO) ewpBotParams
+    let witnessApiServer = mkWitnessAPIServer (convertWitnessHandler unliftIO)
```

Comment:
So again, it just serves there independently of the parameter passed, which is not what we want. Just make the `--witness-listen` optional:
* If it's not present, we don't launch the api
* If it's present and different from educator api, we launch it separately
* Otherwise we launch it at the same port